plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -253,53 +253,63 @@ func gatherWeatherUrl(r io.Reader) (*Status, error) {
 func gatherWeather(acc telegraf.Accumulator, status *Status) {
 	for _, e := range status.List {
 		tm := time.Unix(e.Dt, 0)
-		acc.AddFields(
-			"weather",
-			map[string]interface{}{
-				"cloudiness":   e.Clouds.All,
-				"humidity":     e.Main.Humidity,
-				"pressure":     e.Main.Pressure,
-				"rain":         e.Rain.Rain3,
-				"sunrise":      time.Unix(e.Sys.Sunrise, 0).UnixNano(),
-				"sunset":       time.Unix(e.Sys.Sunset, 0).UnixNano(),
-				"temperature":  e.Main.Temp,
-				"visibility":   e.Visibility,
-				"wind_degrees": e.Wind.Deg,
-				"wind_speed":   e.Wind.Speed,
-			},
-			map[string]string{
-				"city":     e.Name,
-				"city_id":  strconv.FormatInt(e.Id, 10),
-				"country":  e.Sys.Country,
-				"forecast": "*",
-			},
-			tm)
+
+		fields := map[string]interface{}{
+			"cloudiness":   e.Clouds.All,
+			"humidity":     e.Main.Humidity,
+			"pressure":     e.Main.Pressure,
+			"rain":         e.Rain.Rain3,
+			"sunrise":      time.Unix(e.Sys.Sunrise, 0).UnixNano(),
+			"sunset":       time.Unix(e.Sys.Sunset, 0).UnixNano(),
+			"temperature":  e.Main.Temp,
+			"visibility":   e.Visibility,
+			"wind_degrees": e.Wind.Deg,
+			"wind_speed":   e.Wind.Speed,
+		}
+		tags := map[string]string{
+			"city":     e.Name,
+			"city_id":  strconv.FormatInt(e.Id, 10),
+			"country":  e.Sys.Country,
+			"forecast": "*",
+		}
+
+		if len(e.Weather) > 0 {
+			fields["condition_description"] = e.Weather[0].Description
+			fields["condition_icon"] = e.Weather[0].Icon
+			tags["condition_id"] = strconv.FormatInt(e.Weather[0].ID, 10)
+			tags["condition_main"] = e.Weather[0].Main
+		}
+
+		acc.AddFields("weather", fields, tags, tm)
 	}
 }
 
 func gatherForecast(acc telegraf.Accumulator, status *Status) {
-	tags := map[string]string{
-		"city_id":  strconv.FormatInt(status.City.Id, 10),
-		"forecast": "*",
-		"city":     status.City.Name,
-		"country":  status.City.Country,
-	}
 	for i, e := range status.List {
 		tm := time.Unix(e.Dt, 0)
+		fields := map[string]interface{}{
+			"cloudiness":   e.Clouds.All,
+			"humidity":     e.Main.Humidity,
+			"pressure":     e.Main.Pressure,
+			"rain":         e.Rain.Rain3,
+			"temperature":  e.Main.Temp,
+			"wind_degrees": e.Wind.Deg,
+			"wind_speed":   e.Wind.Speed,
+		}
+		tags := map[string]string{
```

Comment:
Move this back out of the loop so it doesn't get allocated each iteration.

Reply:
oops, thanks